traits/trait_list_object.py

```diff
@@ -268,25 +327,22 @@ def __setitem__(self, key, value):
             doesn't match the number of removed elements.
         """
 
+        removed = _removed_items(self, key)
```

Comment:
Nit: Without reading the docstring of `_removed_items`, it is not obvious that this function is lenient. It would be tempting (as it has happened to bad readers like me) to think this function would not suppress index error and then make this false assumption for the rest of this function... Could we make this lenient behaviour more obvious? e.g. renaming it or passing a default explicitly or adding a comment as a reminder?
(On the default `[]`, it could even be a non-subscriptable bogus value for added defensiveness, we don't expect it to be used)

Reply:
`_removed_items(self, key, index_error_return=None)`?

Reply:
Yeah, that would do. :)

Reply:
Changed; I didn't use quite the same wording I suggested here. Instead we have
```
        removed = _removed_items(self, key, return_for_invalid_index=None)
```